### PR TITLE
HPCC-18129 Add ESP call to reset session timer from UI

### PR DESCRIPTION
**Pull request by wangkx**
Signed-off-by: wangkx <kevin.wang@lexisnexis.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

### CODE REVIEW

**Comment by wangkx** [comment]
@RussWhitehead please review.

File: esp/bindings/http/platform/httpservice.cpp [hunk 1]
```diff
@@ -1205,14 +1276,19 @@ EspAuthState CEspHttpServer::authExistingSession(EspAuthRequest& authReq, unsign
...
-        sendMessage(content.str(), (respFormat == ESPSerializationJSON) ? "application/json" : "text/xml");
-        return authTaskDone;
+        if (strieq(authReq.methodName.str(), "get_session_timeout"))
+            createGetSessionTimeoutResponse(content, respFormat, sessionTree);
+        else if (strieq(authReq.methodName.str(), "reset_session_timeout"))
+            resetSessionTimeout(authReq, sessionID, content, respFormat, sessionTree);;
+        if (!content.isEmpty())
+        {
+            sendMessage(content.str(), (respFormat == ESPSerializationJSON) ? "application/json" : "text/xml");
+            return authTaskDone;
```
Comment (RussWhitehead) [inline esp/bindings/http/platform/httpservice.cpp]: In the code you removed the "return authTaskDone" was unconditional, and now it is not. Was this the intent?

File: esp/bindings/http/platform/httpservice.cpp [hunk 2]
```diff
@@ -1205,14 +1276,19 @@ EspAuthState CEspHttpServer::authExistingSession(EspAuthRequest& authReq, unsign
...
     {
         StringBuffer content;
         ESPSerializationFormat respFormat = m_request->queryContext()->getResponseFormat();
-        createGetSessionTimeoutResponse(content, respFormat, sessionTree);
-        sendMessage(content.str(), (respFormat == ESPSerializationJSON) ? "application/json" : "text/xml");
-        return authTaskDone;
+        if (strieq(authReq.methodName.str(), "get_session_timeout"))
+            createGetSessionTimeoutResponse(content, respFormat, sessionTree);
+        else if (strieq(authReq.methodName.str(), "reset_session_timeout"))
+            resetSessionTimeout(authReq, sessionID, content, respFormat, sessionTree);;
```
Comment (RussWhitehead) [inline esp/bindings/http/platform/httpservice.cpp]: trivial, extra semicolon at EOL

File: esp/bindings/http/platform/httpservice.cpp [hunk 3]
```diff
@@ -1185,6 +1185,77 @@ void CEspHttpServer::createGetSessionTimeoutResponse(StringBuffer& resp, ESPSeri
...
+        }
+        else
+        {
+            resp.append("Session timer reset");
+        }
+    }
+    if (format == ESPSerializationJSON)
+    {
+        resp.append(" }");
+        resp.append(" }");
```
Comment (RussWhitehead) [inline esp/bindings/http/platform/httpservice.cpp]: Can these be collapsed into a single append?

**Review by RussWhitehead — COMMENTED** [review]
A  couple minor questions for you

File: esp/bindings/http/platform/httpservice.cpp (hunk 3 above)
Comment (wangkx) [inline esp/bindings/http/platform/httpservice.cpp]: Ok

File: esp/bindings/http/platform/httpservice.cpp (hunk 2 above)
Comment (wangkx) [inline esp/bindings/http/platform/httpservice.cpp]: Will remove.

File: esp/bindings/http/platform/httpservice.cpp (hunk 1 above)
Comment (wangkx) [inline esp/bindings/http/platform/httpservice.cpp]: The "return authTaskDone" was not unconditional. It happens only if the method name is "get_session_timeout". Now, it happends only if the method name is "get_session_timeout" or "reset_session_timout".

**Comment by wangkx** [comment]
@RussWhitehead two append lines are collapsed into a single append and the extra semicolon is removed. Please review.

**Comment by RussWhitehead** [comment]
@wangkx Looks good

**Pull request merged**